roles/openshift_health_checker/test/docker_image_availability_test.py

```diff
@@ -3,26 +3,171 @@
-@pytest.mark.parametrize('task_vars,expected_result', [
-    (
-        dict(
-            openshift=dict(common=dict(
+@pytest.mark.parametrize('deployment_type,is_active', [
+    ("origin", True),
+    ("openshift-enterprise", True),
+    ("enterprise", False),
+    ("online", False),
+    ("invalid", False),
+    ("", False),
+])
+def test_is_active(deployment_type, is_active):
+    task_vars = dict(
+        openshift_deployment_type=deployment_type,
+    )
+    assert DockerImageAvailability.is_active(task_vars=task_vars) == is_active
+
+
+@pytest.mark.parametrize("is_containerized", [
+    True,
+    False,
+])
+def test_all_images_available_locally(is_containerized):
+    def execute_module(module_name, args, task_vars):
+        if module_name == "yum":
+            return {"changed": True}
+
+        assert module_name == "docker_image_facts"
+        assert 'name' in args
+        assert args['name']
+        return {
+            'images': [args['name']],
+        }
+
+    result = DockerImageAvailability(execute_module=execute_module).run(tmp=None, task_vars=dict(
+        openshift=dict(
+            common=dict(
+                service_type='origin',
+                is_containerized=is_containerized,
+            ),
+            docker=dict(additional_registries=["docker.io"]),
+        ),
+        openshift_deployment_type='origin',
+        openshift_release='v3.4',
+        openshift_image_tag='3.4',
+    ))
+
+    assert not result.get('failed', False)
+
+
+@pytest.mark.parametrize("module_failure", [
+    True,
+    False,
+])
+def test_all_images_available_remotely(module_failure):
+    def execute_module(module_name, args, task_vars):
+        return {
+            'docker_image_facts': {'images': [], 'failed': module_failure},
+        }.get(module_name, {'changed': False})
+
+    result = DockerImageAvailability(execute_module=execute_module).run(tmp=None, task_vars=dict(
+        openshift=dict(
+            common=dict(
                 service_type='origin',
                 is_containerized=False,
-            )),
-            openshift_release='v3.5',
-            deployment_type='origin',
-            openshift_image_tag='',  # FIXME: should not be required
+            ),
+            docker=dict(additional_registries=["docker.io", "registry.access.redhat.com"]),
         ),
-        {'changed': False},
+        openshift_deployment_type='origin',
+        openshift_release='3.4',
+        openshift_image_tag='v3.4',
+    ))
+
+    assert not result.get('failed', False)
+
+
+def test_all_images_unavailable():
+    def execute_module(module_name=None, module_args=None, tmp=None, task_vars=None):
+        if module_name == "command":
+            return {
+                'failed': True,
+            }
+
+        return {
+            'changed': False,
+        }
+
+    check = DockerImageAvailability(execute_module=execute_module)
+    actual = check.run(tmp=None, task_vars=dict(
```

Comment:
For unit testing it seems a little complicated/fragile to run the entire check for every test. Then to make changes/debug you have to follow the logic of the entire check "let's see, this will invoke the X, Y, and Z module so I'll have my stub return something different in each case..."

Not saying you have to refactor the tests for this PR, but something to keep in mind for the future.